src/analysisd/alerts/exec.c

```diff
@@ -106,46 +107,145 @@ void OS_Exec(int execq, int *arq, const Eventinfo *lf, const active_response *ar
-                     __crt_ftell,
-                     lf->generated_rule->sigid,
-                     lf->location,
-                     filename ? filename : "-",
-                     extra_args ? extra_args : "-");
+            agent_info *agt_info = NULL;
+            agt_info = get_agent_info(NULL, NULL, ar->agent_id);
+
+            if (!agt_info) {
+                merror("\n Unable to get agent info");
+                return;
+            }
+
+            // agt_info contains "Wazuh vX.X.X", only the last part is needed.
+            char *version = &agt_info->version[7];
+            if(strcmp(version, "4.2.0") > 0){
+
+               // new message in string format
+
+            } else {
+                /* If lf->location start with a ( was generated by remote agent and its
+                * ID is included in lf->location if missing then it must have been
+                * generated by the local analysisd, so prepend a false id tag */
+                if (lf->location[0] == '(') {
+                    snprintf(exec_msg, OS_SIZE_1024,
+                            "%s %c%c%c %s %s %s %s %ld.%ld %d %s %s %s",
+                            lf->location,
+                            (ar->location & ALL_AGENTS) ? ALL_AGENTS_C : NONE_C,
+                            (ar->location & REMOTE_AGENT) ? REMOTE_AGENT_C : NONE_C,
+                            (ar->location & SPECIFIC_AGENT) ? SPECIFIC_AGENT_C : NONE_C,
+                            ar->agent_id != NULL ? ar->agent_id : "(null)",
+                            ar->name,
+                            user,
+                            ip,
+                            (long int)lf->time.tv_sec,
+                            __crt_ftell,
+                            lf->generated_rule->sigid,
+                            lf->location,
+                            filename ? filename : "-",
+                            extra_args ? extra_args : "-");
+                } else {
+                    snprintf(exec_msg, OS_SIZE_1024,
+                            "(local_source) %s %c%c%c %s %s %s %s %ld.%ld %d %s %s %s",
+                            lf->location,
+                            (ar->location & ALL_AGENTS) ? ALL_AGENTS_C : NONE_C,
+                            (ar->location & REMOTE_AGENT) ? REMOTE_AGENT_C : NONE_C,
+                            (ar->location & SPECIFIC_AGENT) ? SPECIFIC_AGENT_C : NONE_C,
+                            ar->agent_id != NULL ? ar->agent_id : "(null)",
+                            ar->name,
+                            user,
+                            ip,
+                            (long int)lf->time.tv_sec,
+                            __crt_ftell,
+                            lf->generated_rule->sigid,
+                            lf->location,
+                            filename ? filename : "-",
+                            extra_args ? extra_args : "-");
+                }
+           }  
         }
 
+        int rc;
```

Comment:
I prefer to declare this variable where used in line 251.

Reply:
Change accepted.

Reply:
Change accepted.